data-transfer/src/main/java/it/eng/datatransfer/service/DataTransferService.java

```diff
@@ -96,7 +97,7 @@ public TransferProcess initiateDataTransfer(TransferRequestMessage transferReque
 				.format(transferRequestMessage.getFormat())
 				.dataAddress(transferRequestMessage.getDataAddress())
 				.state(TransferState.REQUESTED)
-				.role(transferProcess.getRole())
+				.role(IConstants.ROLE_PROVIDER)
```

Comment:
This one is not correctly set when creating TP.initialized?

Reply:
It was, but some tests could have been done with some workaround and passed. With this we are assuring that only ones which make sense can be written in a proper maner.